### DISCUSSION TITLE
Searching for multiple terms or creating aliases

### DISCUSSION THREAD

**Discussion by robbevp** [Pagefind/pagefind#504] [Q&A]
Hi there 👋 

Thanks for this great library!

I've come across a situation where we want to show results for one term, when searching for another one. For example, if a user searches for `AI`, I also want to show results for `Artificial Intelligence`.

I'm wondering if one of the following is possible:
1) Can we create a list of custom aliases for search term?
2) Is there a way to search for two separate options at once? I've tried some common options like `AI|Artificial intelligence`, `AI,Artificial intelligence`, but none of these seem to work.

I've tried some things using the `processTerm` options - either by appending a term, or by replacing it - but it never fully works. In these cases results for `AI` itself might not give a match, since the page might not contain `Artificial intelligence`.

```
processTerm: function (term) {
  return term === "AI" ? `${term} Artificial Intelligence` : term
}
```
```
processTerm: function (term) {
  return term === "AI" ? "Artificial Intelligence" : term
}
```

**Comment by bglw** [comment]
Nothing for these yet! Aliases is a great suggestion, I'll track that as a new issue. Regarding searching multiple terms, #329 is the best place to leave any thoughts 🙂

**Reply by Karlstens** [reply]
The ability to have search aliases is a great idea. Perhaps it could be as simple as the user defining an alias.JSON file?

```json
{
  "example": ["sample", "instance", "case"],
  "case": ["bag", "ticket", "support ID", "instance"],
  "ticket": ["pass", "voucher", "token", "support ID"],
  "bag": ["sack", "pouch", "case", "container"],
  "support ID": ["ticket", "help request number", "service code"],
  "instance": ["example", "occurrence", "case"],
  "container": ["holder", "carrier", "bag"]
}
```

However from an implementation stance, I'm not sure what dragons lie within. 🐉

**Reply by ronan-smith** [reply]
@Karlstens As Pagefind builds the index from existing pages, I feel like it would be more manageable to implement a new metadata key such as a `data-pagefind-meta="alias"` attribute?
